### ISSUE TITLE
HA core 2024.1.0b0 update some errors

### ISSUE THREAD

**Issue by Anto79-ops** [iMicknl/ha-nest-protect#290]
### The problem

Hi, running 0.3.12 and updated HA Core 2024.1.0b0 and there were some warnings in the log for this integration:

```
2023-12-27 15:25:56.309 WARNING (MainThread) [homeassistant.const] TEMP_CELSIUS was used from nest_protect, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfTemperature.CELSIUS instead, please create a bug report at https://github.com/imicknl/ha-nest-protect/issues
```

```
2023-12-27 15:25:56.309 WARNING (MainThread) [homeassistant.const] TEMP_CELSIUS was used from nest_protect, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfTemperature.CELSIUS instead, please create a bug report at https://github.com/imicknl/ha-nest-protect/issues
```

### What version of this integration (ha-nest-protect) has the issue?

0.3.12

### What version of Home Assistant Core has the issue?

2024.1.0

### Device / Model

Nest Protect Battery

### Home Assistant log

<details><summary>Logs</summary>
```
Copy/paste any log here, between the starting and ending backticks (`)
```

</details>

**Comment by poudenes** [comment]
This error is there on the HA update 2024.1.0. Last working HA version without this error was: 2023.12.5

**Comment by iMicknl** [comment]
Thank, we will fix this. This is a warning by the way, not an error.

**Comment by niemyjski** [comment]
@iMicknl any chance we can get this merged?

**Comment by iMicknl** [comment]
Fixed in [v0.4.0b5](https://github.com/iMicknl/ha-nest-protect/releases/tag/v0.4.0b5)
